module_utils/network_lsr/argument_validator.py

```diff
@@ -2216,6 +2216,15 @@ def _ipv6_is_not_configured(connection):
                     idx,
                     "ip.ipv6_disabled is not supported by initscripts.",
                 )
+            if not hasattr(Util.NM(), "SETTING_IP6_CONFIG_METHOD_DISABLED"):
+                raise ValidationError.from_connection(
+                    idx,
+                    "ip.ipv6_disabled is not supported by the running version of "
+                    "NetworkManger. However, if not explicitly enabling "
```

Comment:
After reading this, it is not clear to me why someone would need to use ipv6_disabled at all. What do you expect a user to do if setting ipv6_disabled is not supported.

Reply:
The users need to use `ipv6_disabled` when they want to disable IPv6 on a device, in the situation that `ipv6_disabled` is not supported, user can use `ipv6_ignore` instead (IPv6 is not required or IPv6 is handled by some other mechanism, and NetworkManager should not configure IPv6 for this connection), but this feature has not been implemented by the role. To simulate the `ipv6_ignore` setting, the user can set `ipv6_disabled` to False  (if not specifying, `ipv6_disabled` is set to False by default) and set `auto6` to False (if not specifying, `auto6` is set to True by default) and not specifying any IPv6 address. I got hint from:

```python
            if ip["ipv6_disabled"]:
                s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, "disabled")
            elif ip["auto6"]:
                s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, "auto")
                is_ipv6_configured = True
            elif addrs6:
                s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, "manual")
                is_ipv6_configured = True
            else:
                # we should not set "ipv6.method=ignore". "ignore" is a legacy mode
                # and not really useful. Instead, we should set "link-local" here.
                #
                # But that fix is a change in behavior for the role, so it needs special
                # care.
                s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, "ignore")
```

Reply:
I updated. Thanks.